### PR TITLE
Issue 3307 - V5 Project Selection Screen

### LINKED ISSUE #3307: V5 Project select screen
### Description
Adding Project Select screen to V5. This component will show the projects list for the current teamspace and give the user the ability to navigate to a project url, share that url and delete a project.

### Goals
- [ ] As a User I want to select a project from a project list.

### Tasks
- [x] Add a new component to show the projects list in the route `/v5/dashboard/:teamspace/` 
- [x] Add the layout to show a header component, a tab component and the projects list,  as shown in the mockup.
- [x] Create the header component 
- [x] Create the tabs
- [x] Add routing  for  `/v5/dashboard/:teamspace/t/projects` , `/v5/dashboard/:teamspace/t/settings`,`/v5/dashboard/:teamspace/t/users`  and  `/v5/dashboard/:teamspace/t/jobs`.
- [x] Create the projects list container with its search field and new project menu
- [x] Create the project list and the filtering mechanism
- [x] The individual project card must have an ellipsis menu with the share and delete button that will trigger nothing
- [x] Add the create new project card and button
- [x] Add tests if  some redux/saga files are added.

### Differences with the mockup
- The quota bit in the header component will not be addressed here. 

### Related Resources
[MockUp](https://www.figma.com/file/djgxvRKOaMMyutswn5yywU/Teamspace-(live)?node-id=5323%3A35803)

### PR DESCRIPTION
**Pull request by Amantini1997**
This fixes #3307 

#### Description
- Added the project selection screen;
- Added navigation bar to project selection screen.

### CODE REVIEW

File: frontend/src/v5/ui/routes/dashboard/teamspaces/projects/projectsList.component.tsx
Comment (Amantini1997) [inline frontend/src/v5/ui/routes/dashboard/teamspaces/projects/projectsList.component.tsx:2]: For Daniel,  the Copyright year is correct. This file already existed, but it was moved to a different folder, so GitHub thinks it's a new file.

File: frontend/src/v5/ui/routes/dashboard/teamspaces/projects/projectsList.styles.ts
Comment (Amantini1997) [inline frontend/src/v5/ui/routes/dashboard/teamspaces/projects/projectsList.styles.ts:2]: For Daniel,  the Copyright year is correct. This file already existed, but it was moved to a different folder, so GitHub thinks it's a new file.

File: frontend/src/v5/ui/routes/dashboard/teamspaces/teamspaceLayout/teamspaceLayout.component.tsx
Comment (Amantini1997) [inline frontend/src/v5/ui/routes/dashboard/teamspaces/teamspaceLayout/teamspaceLayout.component.tsx:2]: For Daniel,  the Copyright year is correct. This file already existed, but it was moved to a different folder, so GitHub thinks it's a new file.

File: frontend/src/v5/ui/routes/dashboard/teamspaces/teamspaceLayout/teamspaceLayout.styles.ts
Comment (The-Daniel) [inline frontend/src/v5/ui/routes/dashboard/teamspaces/teamspaceLayout/teamspaceLayout.styles.ts:43]: @george-hadfield will probably bring this up in his review, but the border fades into the gradient background towards the right of the screen. I solved this in the teamspace selection by using white with 10% opacity.

File: frontend/src/v5/ui/components/shared/teamspaceNavigation/teamspaceNavigaton.styles.ts
Comment (The-Daniel) [inline frontend/src/v5/ui/components/shared/teamspaceNavigation/teamspaceNavigaton.styles.ts:17]: This file seems very similar to `projectNavigation.styles.ts`. Is there a way to reuse it?

File: frontend/src/v5/ui/routes/dashboard/teamspaces/teamspaceLayout/teamspaceLayout.styles.ts
Comment (The-Daniel) [inline frontend/src/v5/ui/routes/dashboard/teamspaces/teamspaceLayout/teamspaceLayout.styles.ts:42]: This TODO is not needed

**Review by The-Daniel — DISMISSED** [review]

File: frontend/src/v5/ui/components/shared/teamspaceNavigation/teamspaceNavigation.component.tsx
Comment (sanmont3drepo) [inline frontend/src/v5/ui/components/shared/teamspaceNavigation/teamspaceNavigation.component.tsx:31]: Lets keep the rest of the routing but because in this instance we have only projects defined lets get rid of the rest of the tabs.

**Review by george-hadfield — DISMISSED** [review]
image

Looks great and happy to approve if we can remove the hover and selected states for the ellipsis on the cards.

File: frontend/src/v5/ui/components/shared/linkCard/linkCard.styles.ts
Comment (sanmont3drepo) [inline frontend/src/v5/ui/components/shared/linkCard/linkCard.styles.ts:23]: can we rename this to `CardListItem` ?

Comment (sanmont3drepo) [reply frontend/src/v5/ui/components/shared/linkCard/linkCard.styles.ts:23]: Also Link to CardLink

**Review by sanmont3drepo — CHANGES_REQUESTED** [review]

File: frontend/src/v5/ui/routes/dashboard/index.tsx
Comment (Amantini1997) [reply frontend/src/v5/ui/routes/dashboard/index.tsx:64]: Shall I open a new issue for this?

**Comment by Amantini1997** [comment]
> image
> 
> Looks great and happy to approve if we can remove the hover and selected states for the ellipsis on the cards.

There already is an [issue](https://github.com/3drepo/3drepo.io/issues/3158) for removing the hover and active styling from the ellipsis button, as well as other stuff. that issue was put on hold I think in favour of the V5 viewer facelift.

> ### Update states for ellipsis in all places
> image
> 
> Default - as it is currently Hover - as it is currently Pressed - remove the circle background Active - remove the circle background

File: frontend/src/v5/ui/components/shared/linkCard/linkCard.styles.ts
Comment (Amantini1997) [reply frontend/src/v5/ui/components/shared/linkCard/linkCard.styles.ts:23]: I will proceed to rename ListItem to CardListItem, but I wonder id the second one may be confusing since the component itself is already called `LinkCard` and it would contain a `CardLink` component

File: frontend/src/v5/ui/routes/dashboard/index.tsx
Comment (sanmont3drepo) [reply frontend/src/v5/ui/routes/dashboard/index.tsx:64]: Yup

File: frontend/src/v5/ui/routes/dashboard/teamspaces/teamspaceLayout/teamspaceLayout.styles.ts
Comment (Amantini1997) [reply frontend/src/v5/ui/routes/dashboard/teamspaces/teamspaceLayout/teamspaceLayout.styles.ts]: It was there because the line separating the top bar from the breadcrumbs was not defined. I am now using a PRIMARY_MAIN_CONTRAST at 10% opacity as you suggested

File: frontend/src/v5/services/routing/routing.tsx
Comment (Amantini1997) [reply frontend/src/v5/services/routing/routing.tsx]: This method was deleted in favour of a `projectRoute` method that relies on the `generatePath`

File: frontend/src/v5/ui/components/shared/teamspaceNavigation/teamspaceNavigation.component.tsx
Comment (Amantini1997) [reply frontend/src/v5/ui/components/shared/teamspaceNavigation/teamspaceNavigation.component.tsx]: Done

File: frontend/src/v5/ui/routes/dashboard/index.tsx
Comment (Amantini1997) [reply frontend/src/v5/ui/routes/dashboard/index.tsx:64]: https://github.com/3drepo/3drepo.io/issues/3386

File: frontend/src/v5/ui/routes/dashboard/index.tsx
Comment (Amantini1997) [reply frontend/src/v5/ui/routes/dashboard/index.tsx]: Done

File: frontend/src/v5/ui/components/shared/linkCard/linkCard.styles.ts
Comment (sanmont3drepo) [reply frontend/src/v5/ui/components/shared/linkCard/linkCard.styles.ts:23]: Yeah, I think taking out the Card prefix might be better. These are subcomponents only used for LinkCard, right?

Comment (Amantini1997) [reply frontend/src/v5/ui/components/shared/linkCard/linkCard.styles.ts:23]: Mostly. The `ListItem` is used also by the `AddProjectCard` and `AddTeamspaceCard`, and the `StyledCard` is also used by the `TeamspacePlaceholderCard`, but generally speaking, they all are LinkCards in a sense

**Review by sanmont3drepo — APPROVED** [review]

**Review by sanmont3drepo — APPROVED** [review]

**Review by The-Daniel — DISMISSED** [review]

**Review by george-hadfield — DISMISSED** [review]

**Comment by Amantini1997** [comment]
> image The search bar overflows its container on small screens.

This behaviour is also present in the dashboard project screen to be honest.
image
Note the "all containers" header that falls out of screen. 

The thing is I don't think we support devices which are that small yet (the bug happens at a viewport which is about the Tablet viewport), so we do not have a design for it. 

@george-hadfield what do you suggest for this?

**Review by sanmont3drepo — APPROVED** [review]

**Review by The-Daniel — APPROVED** [review]

**Review by george-hadfield — DISMISSED** [review]

**Review by The-Daniel — APPROVED** [review]

**Pull request merged**
